src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py

```diff
@@ -24,23 +24,34 @@ def storage_copy(cmd, source=None,
                  source_share=None,
                  source_file_path=None,
                  source_local_path=None,
-                 destination_account_name=None,
+                 account_name=None,
                  destination_container=None,
                  destination_blob=None,
                  destination_share=None,
                  destination_file_path=None,
                  destination_local_path=None,
                  exclude_pattern=None, include_pattern=None, exclude_path=None, include_path=None,
-                 follow_symlinks=None):
-    def get_url_with_sas(source, account_name, container, blob, share, file_path, local_path):
+                 follow_symlinks=None, **kwargs):
+    def get_url_with_sas(source, account_name, container, blob, share, file_path, local_path, **kwargs):
         import re
         import os
         from azure.cli.command_modules.storage._validators import _query_account_key
         from azure.cli.command_modules.storage.azcopy.util import _generate_sas_token
         storage_endpoint = cmd.cli_ctx.cloud.suffixes.storage_endpoint
+        account_key = kwargs.pop('account_key', None)
+        connection_string = kwargs.pop('connection_string', None)
+        sas_token = kwargs.pop('sas_token', None)
```

Comment:
what's the expected behavior if all the three arguments are provided?

Reply:
Sync with Zunli offline: there is pre logic before going to this method that handles these arguments and only one of them will be left.

---
In reply to: [470446058](https://github.com/Azure/azure-cli/pull/14636#discussion_r470446058) [](ancestors = 470446058)